packages/ra-ui-materialui/src/input/BooleanInput.js

```diff
@@ -51,6 +53,13 @@ export class BooleanInput extends Component {
                         />
                     }
                 />
+                {meta.error
+                    ? meta.error.map((error, index) => (
```

Comment:
`meta.error` is a String, not an array (see https://redux-form.com/8.2.0/docs/api/field.md/#-code-meta-error-string-code-optional)